hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/spark/sql/hudi/ddl/TestAlterTable.scala

```diff
@@ -371,11 +375,9 @@ class TestAlterTable extends HoodieSparkSqlTestBase {
         val newLocation2 = spark.sessionState.catalog.getTableMetadata(new TableIdentifier(newLocTableName))
           .properties.get("path")
         if (oldLocation2.nonEmpty) {
-          // Remove the impact of the schema.
-          val oldLocation2Path = new Path(oldLocation2.get.stripPrefix("file:"))
-          val newLocation2Path = new Path(newLocation2.get.stripPrefix("file:"))
+          // the scheme and authority need to match as well
           assertResult(true)(
-            newLocation2Path.equals(oldLocation2Path)
+            oldLocation2.get.equals(newLocation2.get)
```

Comment:
This is the key change in test, we want to ensure the path won't change even with the scheme